android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/authorization/DbContentProvider.java

```diff
@@ -1,56 +1,60 @@
+/*
+ * Copyright (c) 2022 Solana Mobile Inc.
+ */
+
 package com.solana.mobilewalletadapter.walletlib.authorization;
 
 import android.content.ContentValues;
 import android.database.Cursor;
 import android.database.sqlite.SQLiteDatabase;
 
-public abstract class DbContentProvider<T> {
-    public SQLiteDatabase mDb;
+/*package*/ abstract class DbContentProvider<T> {
+    private final SQLiteDatabase mDb;
 
-    public DbContentProvider(SQLiteDatabase db) {
+    /*package*/ DbContentProvider(SQLiteDatabase db) {
         this.mDb = db;
     }
 
-    protected abstract T cursorToEntity(Cursor cursor);
+    /*package*/ abstract T cursorToEntity(Cursor cursor);
 
-    public long insert(String tableName, ContentValues values) {
-        return mDb.insert(tableName,null, values);
+    /*package*/ long insert(String tableName, ContentValues values) {
```

Comment:
MAYBE: should `insert`, `delete`, etc be package-visible or `protected`? I worry that making them visible outside of subclasses encourages someone to accidentally call them directly, and we lose the benefit of encapsulating the table inside of a Dao.